src/combineReducers.js

```diff
@@ -90,50 +140,20 @@ function assertReducerSanity(reducers) {
  * @returns {Function} A reducer function that invokes every reducer inside the
  * passed object, and builds a state object with the same shape.
  */
-export default function combineReducers(reducers) {
-  var reducerKeys = Object.keys(reducers)
-  var finalReducers = {}
-  for (var i = 0; i < reducerKeys.length; i++) {
-    var key = reducerKeys[i]
-    if (typeof reducers[key] === 'function') {
-      finalReducers[key] = reducers[key]
-    }
-  }
-  var finalReducerKeys = Object.keys(finalReducers)
 
-  var sanityError
-  try {
-    assertReducerSanity(finalReducers)
-  } catch (e) {
-    sanityError = e
-  }
+export default function combineReducers(reducers) {
+  var fr = finalReducers(reducers)
+  var keys = finalReducerKeys(fr)
 
   return function combination(state = {}, action) {
-    if (sanityError) {
-      throw sanityError
+    if (sanityError(reducers)) {
```

Comment:
This looks concerning.  The previous code only ran `assertReducerSanity()` once, but it looks like the modified code is actually calling this twice.